scrapy/core/spidermw.py

```diff
@@ -58,7 +60,9 @@ def _process_spider_input(self, scrape_func: ScrapeFunc, response: Response, req
                 return scrape_func(Failure(), request, spider)
         return scrape_func(response, request, spider)
 
-    def _evaluate_iterable(self, response, spider, iterable, exception_processor_index, recover_to):
+    def _evaluate_iterable(self, response: Response, spider: Spider, iterable: Union[Iterable, AsyncIterable],
+                           exception_processor_index: int, recover_to: Union[MutableChain, MutableAsyncChain]
```

Comment:
I wonder if it would make sense to provide in `scrapy.utils.typing` aliases for some of these sync-or-async types (e.g. `AnyIterable`, `AnyMutableChain`). Although maybe it’s more readable to specify them explicitly with `Union` as we do now, instead of having to go to a different module to find out what the type stands for.